src/serverSetup.ts

```diff
@@ -35,28 +35,69 @@ export class ServerInstallError extends Error {
     }
 }
 
-export async function installCodeServer(conn: SSHConnection, serverDownloadUrlTemplate: string, extensionIds: string[], envVariables: string[], useSocketPath: boolean, logger: Log): Promise<ServerInstallResult> {
+export async function installCodeServer(conn: SSHConnection, serverDownloadUrlTemplate: string, extensionIds: string[], envVariables: string[], platform: string | undefined, useSocketPath: boolean, logger: Log): Promise<ServerInstallResult> {
+    if (!platform) {
+        const detectedPlatform = await conn.exec('uname -s');
+        if (detectedPlatform.stdout.includes('MINGW64')) {
+            platform = 'windows'
+        } else if (detectedPlatform.stdout.includes('windows32')) {
+            platform = 'windows'
+        }
+        
+        if (platform) {
+            logger.trace(`Detected platform: ${platform}`);
+        }
+    }
+
     const scriptId = crypto.randomBytes(12).toString('hex');
 
     const vscodeServerConfig = await getVSCodeServerConfig();
-    const installServerScript = generateServerInstallScript({
-        id: scriptId,
-        version: vscodeServerConfig.version,
-        commit: vscodeServerConfig.commit,
-        quality: vscodeServerConfig.quality,
-        release: vscodeServerConfig.release,
-        extensionIds,
-        envVariables,
-        useSocketPath,
-        serverApplicationName: vscodeServerConfig.serverApplicationName,
-        serverDataFolderName: vscodeServerConfig.serverDataFolderName,
-        serverDownloadUrlTemplate,
-    });
-
-    logger.trace('Server install command:', installServerScript);
-    // Fish shell does not support heredoc so let's workaround it using -c option,
-    // also replace single quotes (') within the script with ('\'') as there's no quoting within single quotes, see https://unix.stackexchange.com/a/24676
-    const commandOutput = await conn.exec(`bash -c '${installServerScript.replace(/'/g, `'\\''`)}'`);
+    
+    let commandOutput: { stdout: string; stderr: string };
+    if (platform == 'windows') {
+         const installServerScript = generatePowerShellInstallScript({
+            id: scriptId,
+            version: vscodeServerConfig.version,
+            commit: vscodeServerConfig.commit,
+            quality: vscodeServerConfig.quality,
+            release: vscodeServerConfig.release,
+            extensionIds,
+            envVariables,
+            useSocketPath,
+            serverApplicationName: vscodeServerConfig.serverApplicationName,
+            serverDataFolderName: vscodeServerConfig.serverDataFolderName,
+            serverDownloadUrlTemplate,
+        });
+        
+        logger.trace('Server install command:', installServerScript);
+
+        const installDir = `$HOME\\${vscodeServerConfig.serverDataFolderName}\\install`;
+        const installScript = `${installDir}\\${vscodeServerConfig.commit}.ps1`;
+        const endRegex = new RegExp(`${scriptId}: end`);
```

Comment:
@daiyam  why we need this reges test?

Reply:
Yeah! I forget to leave a comment about that. 
I haven't found a way to fork the server to a new process. So I keep the current script running with `while($True)`/`sleep 30` at the end.
By doing so, I need to the result details to be detected in the output (since the command execution never finish) which is done with `endRegex` and the method `execPartial`.

Reply:
I see, makes sense then